### PR TITLE
adds command for hard-deleting soft-deleted tokens

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds command for hard-deleting soft-deleted tokens

## Why are we making these changes?

We would like to clean up soft-deleted tokens from waiter clusters.

### CODE REVIEW

File: token-syncer/src/token_syncer/utils.clj [hunk 1]
```diff
@@ -13,9 +13,27 @@
...
 
 (defn successful?
   "Returns true if the response has a 2XX status code."
   [{:keys [status]}]
   (and (integer? status) (<= 200 status 299)))
+
+(defn iso8601->millis
+  "Convert the ISO 8601 string to numeric milliseconds."
+  [date-str]
+  (-> (:date-time f/formatters)
```
Comment (sradack) [inline token-syncer/src/token_syncer/utils.clj]: Nitpick: These threading calls are awkward.  For me, threading makes the most sense when you start with input data and go through a series of transformations.

File: token-syncer/src/token_syncer/commands/cleanup.clj [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                                              (- (System/currentTimeMillis) (-> 1 t/weeks t/in-millis)))
+                                cluster-url (first arguments)]
+                            (log/info "hard-deleting tokens from" cluster-url "soft-deleted before"
+                                      (utils/millis->iso8601 before-time))
+                            (cleanup-tokens waiter-api cluster-url before-time)
+                            {:exit-code 0
+                             :message "exiting with code 0"}))))
+   :option-specs [["-b" "--before"
+                   (str "Hard-delete tokens soft-deleted before the specified time. "
+                        "Must be specified in as an epoch time. "
```
Comment (sradack) [inline token-syncer/src/token_syncer/commands/cleanup.clj]: Should we use iso8601 as the input format?

File: token-syncer/integration/token_syncer/basic_test.clj [hunk 3]
```diff
@@ -840,3 +841,38 @@
...
+
+      (try
+        ;; ACT
+        (let [actual-result (cleanup/cleanup-tokens waiter-api cluster-url current-time-ms)]
+
+          ;; ASSERT
+          (is (= #{token-name} actual-result))
+          (is (= {:description {}
+                  :headers {"content-type" "application/json"
+                            "etag" ""}
```
[the remark this answers is not in the document]
Comment (shamsimam) [reply token-syncer/integration/token_syncer/basic_test.clj]: Fixed, empty etags are no longer returned on 404 responses.

File: token-syncer/src/token_syncer/commands/cleanup.clj (hunk 2 above)
Comment (shamsimam) [reply token-syncer/src/token_syncer/commands/cleanup.clj]: Changed to use iso8601 as input format.

File: token-syncer/src/token_syncer/utils.clj (hunk 1 above)
Comment (shamsimam) [reply token-syncer/src/token_syncer/utils.clj]: Changed.

**Pull request merged**